tests/publiccloud/xfsprepare.pm

```diff
@@ -0,0 +1,79 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Prepare secondary disk for usage in xfstests
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use version_utils qw(is_sle);
+use utils;
+
+# xfstests configuration file, as required by xfstests/run
+my $INST_DIR = '/opt/xfstests';
+my $CONFIG_FILE = "$INST_DIR/local.config";
+
+# Install requirements
+sub install_xfstests {
+    # Add filesystem repository, which contains the xfstests
+    assert_script_run('zypper -n up', timeout => 300);
+    assert_script_run('zypper ar https://download.opensuse.org/repositories/filesystems/SLE_15_SP3/ filesystems');
+    assert_script_run('zypper --gpg-auto-import-keys -n ref');
+    zypper_call('in xfsprogs xfsdump btrfsprogs kernel-default xfstests');
+    script_run('ln -s /usr/lib/xfstests/ /opt/xfstests');    # xfstests/run expects the tests to be in /opt/xfstests
```

Comment:
I have no good reason why, I just thought it is not necessary. Following the "fail early" principle, it might be better to use `assert_script_run` here as well.